CollapseLauncher/XAMLs/MainApp/MainWindow.xaml.cs

```diff
@@ -178,12 +178,62 @@ public void InitializeWindowSettings()
         private IntPtr WndProc(IntPtr hwnd, uint msg, UIntPtr wParam, IntPtr lParam)
         {
             const uint WM_SYSCOMMAND = 0x0112;
-            const uint SC_MAXIMIZE = 0xF030;
-            if (msg == WM_SYSCOMMAND && wParam == SC_MAXIMIZE)
+            const uint WM_SHOWWINDOW = 0x0018;
+            const uint WM_SIZE = 0x0005;
+            switch (msg)
             {
-                // Ignore WM_SYSCOMMAND SC_MAXIMIZE message
-                // Thank you Microsoft :)
-                return 1;
+                case WM_SYSCOMMAND:
+                {
+                    const uint SC_MAXIMIZE = 0xF030;
+                    const uint SC_MINIMIZE = 0xF020;
+                    switch (wParam)
+                    {
+                        case SC_MAXIMIZE:
+                        {
+                            // Ignore WM_SYSCOMMAND SC_MAXIMIZE message
+                            // Thank you Microsoft :)
+                            return 0;
+                        }
+                        case SC_MINIMIZE:
```

Comment:
Tested this and everything seems okay but the Minimize to Tray function

This case is only hit when user minimize Collapse window through the taskbar, when pressing the usual minimize button, the SC_MINIMIZE did not get triggered.

Reply:
yes, too bad of m_presenter. new code pushed